File: 5. String/multiplyStrings.js
/**
 * @param {string} num1
 * @param {string} num2
 * @return {string}
 */
var multiply = function(num1, num2) {
    let digits = Array(num1.length + num2.length).fill(0);
    let position = digits.length - 1;
    for (let i = num1.length - 1; i >= 0; i--) {
        tempPos = position;
        for (let j = num2.length - 1; j >= 0; j--) {
            digits[tempPos] += num1[i] * num2[j];
            digits[tempPos - 1] += Math.trunc(digits[tempPos] / 10);
            digits[tempPos] %= 10;
            tempPos -= 1;
        }
        position -= 1;
    }

    let zeroPaddingStartIdx = 0;
    while (zeroPaddingStartIdx < digits.length - 1 && digits[zeroPaddingStartIdx] === 0) {zeroPaddingStartIdx++;}
    return digits.slice(zeroPaddingStartIdx).join('');


};

File: 5. String/multiply_strings.py
class Solution:
    def multiply(self, num1, num2):
        product = [0] * len(num1 + num2)
        position = len (product) - 1
        for n1 in num1[::-1]:
            temp = position
            for n2 in num2[:: -1]: # top integer
                int_n1 =ord(n1) - ord('0')
                int_n2 =ord(n2) - ord('0')
                product[temp] += int_n1 * int_n2
                product[temp - 1] += product[temp] // 10
                product[temp] %= 10
                temp -= 1
            position -= 1
        cut = 0
        while cut < len(product) - 1 and product[cut] == 0:
            cut += 1

        return ''.join(map(str, product[cut:]))


